paz/optimization/losses/frustumpointnet_loss.py

```diff
@@ -0,0 +1,104 @@
+import numpy as np
+import tensorflow as tf
+
+from paz.models.detection.model_util import NUM_HEADING_BIN, NUM_SIZE_CLUSTER, g_mean_size_arr, \
+    get_box3d_corners_helper, get_box3d_corners
+
+
+def huber_loss(error, delta):
+    abs_error = tf.abs(error)
+    quadratic = tf.minimum(abs_error, delta)
+    linear = (abs_error - quadratic)
+    losses = 0.5 * quadratic ** 2 + delta * linear
+    return tf.reduce_mean(losses)
+
+
+def FPointNet_loss(args, corner_loss_weight=10.0, box_loss_weight=1.0, mask_weight=1.0):
+    """ Loss functions for 3D object detection.
+    Input:
+        args: dict
+                mask_label: TF int32 tensor in shape (B,N)
+                center_label: TF tensor in shape (B,3)
+                heading_class_label: TF int32 tensor in shape (B,)
+                heading_residual_label: TF tensor in shape (B,)
+                size_class_label: TF tensor int32 in shape (B,)
+                size_residual_label: TF tensor tensor in shape (B,)
+                end_points: dict, outputs from our model
+        corner_loss_weight: float scalar
+        box_loss_weight: float scalar
```

Comment:
these are too many inputs for a single loss. Try to separate the loss into simple functions that try to take (y_true, y_pred) inputs.